pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1453,17 +1500,21 @@ func doAppNetworkModifyAllUnderlayNetworks(
 	config types.AppNetworkConfig,
 	oldConfig types.AppNetworkConfig,
 	status *types.AppNetworkStatus,
-	ipsets []string) {
+	ipsets []string) error {
 
 	for i := range config.UnderlayNetworkList {
 		log.Tracef("handleModify ulNum %d\n", i)
 		ulConfig := &config.UnderlayNetworkList[i]
 		oldulConfig := &oldConfig.UnderlayNetworkList[i]
 		ulStatus := &status.UnderlayNetworkList[i]
-		doAppNetworkModifyUnderlayNetwork(
-			ctx, status, ulConfig, oldulConfig, ulStatus, ipsets, false)
+		err := doAppNetworkModifyUnderlayNetwork(
+			ctx, status, ulConfig, oldulConfig, ulStatus, ipsets, i, false)
+		if err != nil {
+			return err
```

Comment:
Missing publish call here? Ah - addError() does a secret publish so it isn't needed.